### ISSUE TITLE
How to define a list of <class-name/> elements in brokerProperties

### ISSUE THREAD

**Issue by braveadmin** [artemiscloud/activemq-artemis-operator#1062]
I'm trying to define a core configuration through brokerProperties, as defining through the Custom Resource Definition is deprecated (specified [here](https://github.com/artemiscloud/activemq-artemis-operator/blob/1.2.7/deploy/crds/broker_activemqartemis_crd.yaml)).

Concretely, the XML property I'm trying to define is the following:
`<remoting-incoming-interceptors><class-name>my.Own.Interceptor</class-name></remoting-incoming-interceptors>`

As documented [here](https://activemq.apache.org/components/artemis/documentation/2.37.0/configuration-index.html#broker-properties), in order to put this in the brokerProperties, it has to be converted to camelCase Java naming convention, which would be the following:
`remotingIncomingInterceptors.className=my.Own.Interceptor`

However, I'm finding that this is an unknown property:

> some properties from artemis-props resulted in error on pod artemis-ss-0: reasons: map[broker.properties [{"value":"remotingIncomingInterceptors.classNames=my.Own.Interceptor","reason":"java.lang.NoSuchMethodException: Unknown property 'remotingIncomingInterceptors' on class 'class org.apache.activemq.artemis.core.config.impl.FileConfiguration'"}]]

What am I doing wrong? 
I'm finding the documentation absolutely misleading, explaining how to do things in a deprecated way and keeping examples and resources that are no longer required.

**Comment by brusdev** [comment]
@braveadmin just to avoid confusion the `remoting-incoming-interceptors` are not deprecated and are not address settings. As documented [here](https://activemq.apache.org/components/artemis/documentation/2.37.0/configuration-index.html#broker-properties), the configuration properties are low level, lower level than xml, however it is very powerful; any accessible attribute of the internal org.apache.activemq.artemis.core.config.impl.ConfigurationImpl objects can be modified. This means that the attributes of the internal org.apache.activemq.artemis.core.config.impl.ConfigurationImpl class have to be converted to camelCase Java naming convention, i.e. the broker properties for the incoming interceptor class names are:
```
incomingInterceptorClassNames=my.Own.Interceptor1,my.Own.Interceptor2
```

**Comment by braveadmin** [comment]
> ribute of the internal org.apache.activemq.artemis.core.config.impl.ConfigurationImpl object

I already tried this approach and it is not working to me either.

**Comment by braveadmin** [comment]
I have modified the bug description as it may not be enough concise. I was trying to define the core configuration `remoting-incoming-interceptors`. I'm aware that it is not an address-settings, apologies for the confusion and thank you for the approach. I was specifying the class in a wrong way, as the Artemis [documentation](https://activemq.apache.org/components/artemis/documentation/2.37.0/intercepting-operations.html#configuring-the-interceptors) tells to specify the property in broker.xml as `<remoting-incoming-interceptors>` but the class name is different as you said: `incomingInterceptorClassNames`.

Jumping back to the other topic I mixed with this one, I find the Artemis documentation a bit entangled, as there is a lot of properties for addresses that are now deprecated in favour of brokerProperties. Why is the purpose of this approach? I defining these properties in a CRD is way easier and tidier than putting in a brokerProperties, Thanks for your help @brusdev
